src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlighterViewTaggerProvider.cs

```diff
@@ -76,12 +76,13 @@ protected override ITaggerEventSource CreateEventSource(ITextView textView, ITex
             var position = caretPosition.Value;
             var snapshot = snapshotSpan.Snapshot;
 
-            var existingTags = context.GetExistingTags(new SnapshotSpan(snapshot, position, 0));
+            // See if the user is just moving their caret around in an existing tag.  If so, we don't
+            // want to actually go recompute things.  Note: this only works for containment.  If the
+            // user moves their caret to the end of a highlighted reference, we do want to recompute
+            // as they may now be at the start of some other reference that should be highlighted instead.
+            var existingTags = context.GetExistingContainingTags(new SnapshotPoint(snapshot, position));
             if (!existingTags.IsEmpty())
             {
-                // We already have a tag at this position.  So the user is moving from one highlight
-                // tag to another.  In this case we don't want to recompute anything.  Let our caller
-                // know that we should preserve all tags.
                 context.SetSpansTagged(SpecializedCollections.EmptyEnumerable<DocumentSnapshotSpan>());
```

Comment:
see comment later on for why this logic changed.  Specifically, a highlight-refs scenario with regexes identified an interesting problem with how we do highlighting currently (but which you cannot really experience in C#/VB).